File: sample/OpenCvSharp.Samples/Samples/ConvertToWriteableBitmap.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Media;
using System.Windows.Media.Imaging;
using System.Windows.Threading;
using OpenCvSharp;
using OpenCvSharp.Extensions;

namespace OpenCvSharp.Test
{
    /// <summary>
    /// System.Windows.Media.Imaging.WriteableBitmapへの変換
    /// </summary>
    class ConvertToWriteableBitmap
    {
        public ConvertToWriteableBitmap()
        {
            WriteableBitmap wb = null;

            // OpenCVによる画像処理 (Threshold)
            using (IplImage src = new IplImage(Const.ImageLenna, LoadMode.GrayScale))
            using (IplImage dst = new IplImage(src.Size, BitDepth.U8, 1))
            {
                src.Smooth(src, SmoothType.Gaussian, 5);
                src.Threshold(dst, 0, 255, ThresholdType.Otsu);
                // IplImage -> WriteableBitmap
                wb = dst.ToWriteableBitmap(PixelFormats.BlackWhite);
                //wb = WriteableBitmapConverter.ToWriteableBitmap(dst, PixelFormats.BlackWhite);
            }

            // WPFのWindowに表示してみる
            Image image = new Image { Source = wb };
            Window window = new Window
            {
                Title = "from IplImage to WriteableBitmap",
                Width = wb.PixelWidth,
                Height = wb.PixelHeight,
                Content = image
            };

            Application app = new Application();
            app.Run(window);
        }
    }
}

File: sample/CStyleSamplesVB/Samples/ConvertToWriteableBitmap.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Media
Imports System.Windows.Media.Imaging
Imports System.Windows.Threading
Imports OpenCvSharp
Imports OpenCvSharp.Extensions

' Namespace OpenCvSharpSamplesVB
Imports SampleBase

''' <summary>
''' System.Windows.Media.Imaging.WriteableBitmapへの変換
''' </summary>
Friend Module ConvertToWriteableBitmap
    Public Sub Start()
        Dim wb As WriteableBitmap = Nothing

        ' OpenCVによる画像処理 (Threshold)
        Using src As New IplImage(FilePath.Image.Lenna, LoadMode.GrayScale)
            Using dst As New IplImage(src.Size, BitDepth.U8, 1)
                src.Smooth(src, SmoothType.Gaussian, 5)
                src.Threshold(dst, 0, 255, ThresholdType.Otsu)
                ' IplImage -> WriteableBitmap
                wb = dst.ToWriteableBitmap(PixelFormats.BlackWhite)
                'wb = WriteableBitmapConverter.ToWriteableBitmap(dst, PixelFormats.BlackWhite);
            End Using
        End Using

        ' WPFのWindowに表示してみる
        Dim image As Image = New Image With {.Source = wb}
        Dim window As Window = New Window With {.Title = "from IplImage to WriteableBitmap", .Width = wb.PixelWidth, .Height = wb.PixelHeight, .Content = image}

        Dim app As New Application()
        app.Run(window)
    End Sub
End Module
' End Namespace
